src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -1294,17 +1294,27 @@ void GCToOSInterface::GetMemoryStatus(uint64_t restricted_limit, uint32_t* memor
         }
         else
         {
-            available = GetAvailablePhysicalMemory();
-
             if (memory_load != NULL)
```

Comment:
Judging by the comment on line 1271, this branch won't be used where .NET is running in a container. If we're in a container with a cgroup memory limit, `restricted_limit` would be set and so we wouldn't be in this `else` branch.

Reply:
From my understanding of the code `restricted_limit` is set, only if `GCTotalPhysicalMemory` is set in the config. Maybe I am still missing something...

Reply:
My understanding is that `restricted_limit`, if not set by `GCTotalPhysicalMemory` GC config, comes from GCToOSInterface::GetPhysicalMemoryLimit (gcenv.unix.cpp). See the call in GetPhysicalMemoryLimit in GCHeap::Initialize. This in turn calls `GetRestrictedPhysicalMemoryLimit`, which calls through to code that parses the cgroup.